api/rds.proto

```diff
@@ -279,6 +282,12 @@ message RouteMatch {
   // the request with the same values (or based on presence if the value field
   // is not in the config).
   repeated HeaderMatcher headers = 6;
+
+  // [#not-implemented-hide:] Specifies a set of URL query parameters on which the route should
+  // match. The router will check the query string from the *path* header against all the specified
+  // query args. If the number of specified query args is nonzero, they all must match the *path*
+  // header's query string in order or a match to occur.
```

Comment:
Nit: s/or/for/

Does forcing order here make sense? Don't query strings have a dictionary data model?

Reply:
Query strings don't inherently have a dictionary data model, in the sense that RFC 3986 doesn't prohibit duplicate keys -- in fact, it doesn't even specify that the query string must contain key=value data.  But in this diff, I went with the philosophy that order doesn't matter, since HTTP implementors tend to treat the query string as a dictionary.

Reply:
Right, I think actually the above comment is a result of English language ambiguity "in order for a match to occur" could be ready as "they must match in order, for a match to occur". Maybe just reword it slightly.